code/game/verbs.dm

```diff
@@ -96,7 +121,52 @@
 		return
 	if(!tgui_only)
 		verbs -= old_verbs
+	remove_verbs(old_verbs)
 	return ..(old_verbs)
 
+/mob/remove_verb(old_verbs, tgui_only = FALSE)
+	if(IsAdminAdvancedProcCall())
+		message_admins("[key_name(usr)] attempted to edit their verbs.")
+		log_game("[key_name(usr)] attempted to edit their verbs.")
+		return
+	if(client)
+		client.remove_verbs(old_verbs)
+	return ..()
+
+/mob/add_verb(new_verbs, tgui_only)
+	if(IsAdminAdvancedProcCall())
+		message_admins("[key_name(usr)] attempted to edit their verbs.")
+		log_game("[key_name(usr)] attempted to edit their verbs.")
+		return
+	if(client)
+		client.add_verbs(new_verbs)
+	return ..()
+
+/client/proc/remove_verbs(old_verbs)
+	if(!islist(old_verbs))
+		old_verbs = list(old_verbs)
+	var/list/removed_verbs = list()
+	for(var/procpath/P as() in old_verbs)
+		if(!islist(removed_verbs[P.category]))
+			removed_verbs[P.category] = list()
+		removed_verbs[P.category] += "[P.name]"
+	tgui_panel.remove_verbs(removed_verbs)
+
+/client/proc/add_verbs(new_verbs)
+	if(!islist(new_verbs))
+		new_verbs = list(new_verbs)
+	var/list/added_verbs = list()
+	for(var/procpath/P as() in new_verbs)
+		if(P.invisibility > mob.see_invisible)
```

Comment:
![image](https://user-images.githubusercontent.com/5714543/126916115-0493a460-c24b-4492-a0c0-6f6cd818d52a.png)

Reply:
Lot of these runtimes

Reply:
oh im stupid, its the mob thats null

Reply:
whoops I accidentally editted your comment instead of adding a new one